mne/decoding/base.py

```diff
@@ -749,3 +749,54 @@ def _make_scorer(scoring):
         return get_scorer(scoring)
     else:
         return make_scorer(scoring)
+
+
+def _get_final_est(estimator):
+    """Return the final component of a sklearn estimator/pipeline.
+
+    Parameters
+    ----------
+    estimator : pipeline, or sklearn / MNE estimator
+        An estimator chain from which you wish to pull the final estimator.
+
+    Returns
+    -------
+    estimator : the sklearn-style estimator at the end of the input chain.
+    """
+    # Define classes where we pull `estimator` manually
+    import sklearn
+    from distutils.version import LooseVersion
+    if LooseVersion(sklearn.__version__) < '0.16':
+        raise ValueError('sklearn support requires sklearn version >= 0.16')
+
+    attributes = [ii for ii in ['_final_estimator', 'base_estimator']
+                  if hasattr(estimator, ii)]
+    if len(attributes) > 0:
+        # In the event that both are present, `_final_estimator` is prioritized
+        estimator = getattr(estimator, attributes[0])
+        estimator = _get_final_est(estimator)
+    return estimator
+
+
+def get_coefs(estimator, coef_name='coef_'):
+    """Pull coefficients from an estimator object.
+
+    Parameters
+    ----------
+    estimator : a sklearn estimator
+        The estimator from which to pull the coefficients.
+    coef_name : string
+        The name of the attribute corresponding to the coefficients created
+        after fitting. Defaults to 'coef_'
+
+    Returns
+    -------
+    coefs : array, shape (n_targets, n_coefs)
+        The output coefficients.
+    """
+    estimator = _get_final_est(estimator)
```

Comment:
IMO the function should be _get_coefs, and incorporate the lines below.

I routinely use a similar function with an adding option to invert_transform the linear coefficients. This can be useful if the  estimator is a pipeline, and typically contains a PCA or a scaler